backend/src/handlers_prelude/repo_fs.rs

```diff
@@ -0,0 +1,284 @@
+//! Endpoints for interacting with the repository's filesystem (create doc/asset, read doc/asset, et cetera)
+use crate::git::INode;
+use axum::{
+    body::Bytes,
+    debug_handler,
+    extract::{DefaultBodyLimit, Path, Query, State},
+    http::{HeaderMap, StatusCode},
+    response::IntoResponse,
+    routing::get,
+    Json, Router,
+};
+use reqwest::header::{CONTENT_DISPOSITION, CONTENT_TYPE};
+use serde::{Deserialize, Serialize};
+use tracing::{error, warn};
+
+use crate::{perms::Permission, require_perms, AppState};
+
+use super::eyre_to_axum_err;
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct GetDocQuery {
+    pub path: String,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct GetDocResponse {
+    pub contents: String,
+}
+
+/// This handler accepts a `GET` request to `/api/doc?path=`.
+/// TODO: refactor to pass it in directly as a url path instead of doing the whole url arguments thing
+pub async fn get_doc_handler(
+    State(state): State<AppState>,
+    Query(query): Query<GetDocQuery>,
+) -> Result<Json<GetDocResponse>, (StatusCode, &'static str)> {
+    match state.git.get_doc(&query.path) {
+        Ok(maybe_doc) => maybe_doc.map_or(
+            Err((
+                StatusCode::NOT_FOUND,
+                "The file at the provided path was not found.",
+            )),
+            |doc| Ok(Json(GetDocResponse { contents: doc })),
+        ),
+        Err(e) => {
+            warn!(
+                "Failed to fetch doc with path: {:?}; error: {:?}",
+                query.path, e
+            );
+            Err((
+                StatusCode::INTERNAL_SERVER_ERROR,
+                "Fetch failed, check server logs for more info",
+            ))
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct PutDocRequestBody {
+    contents: String,
+    path: String,
+    commit_message: String,
+}
+
+#[debug_handler]
+pub async fn put_doc_handler(
+    State(state): State<AppState>,
+    headers: HeaderMap,
+    Json(body): Json<PutDocRequestBody>,
+) -> Result<StatusCode, (StatusCode, String)> {
+    // Unauthorized means that the client has not identified itself, where as forbidden
+    // means that the client has identified itself, and it does not have the required permissions
```

Comment:
This seems to be need to be fixed, what means that?

Reply:
Left behind from a refactor